File: rust-sass-libsass-tests/ecosystem/smokes/java/Smoke.java
import io.bit3.jsass.Compiler;
import io.bit3.jsass.Options;
import io.bit3.jsass.Output;

public class Smoke {
  public static void main(String[] args) throws Exception {
    String version = Compiler.getLibsassVersion();
    if (!"3.6.6".equals(version)) {
      throw new IllegalStateException("libsass version: " + version);
    }
    Compiler compiler = new Compiler();
    Output out = compiler.compileString("$color: red; .foo { color: $color; }", new Options());
    String css = out.getCss();
    if (!css.contains(".foo") || !css.contains("color: red") || css.contains("$color")) {
      throw new IllegalStateException("got: " + css);
    }
    System.out.println("java SMOKE-OK");
  }
}

File: rust-sass-libsass-tests/ecosystem/smokes/ruby/smoke.rb
# frozen_string_literal: true

require "sassc"

raise SassC::Native.version.inspect unless SassC::Native.version == "3.6.6"
out = SassC::Engine.new('$color: red; .foo { color: $color; }').render
raise out.inspect unless out == ".foo {\n  color: red; }\n"
puts "ruby SMOKE-OK"
